torch/csrc/autograd/functions/accumulate_grad.h

```diff
@@ -84,7 +130,22 @@ struct TORCH_API AccumulateGrad : public Node {
         if (new_grad.is_sparse()) {
           update_grad(new_grad.clone());
         } else {
-          update_grad(new_grad.clone(at::MemoryFormat::Contiguous));
+          // Deep copies new_grad according to the "Gradient Layout Contract."
+          if (variable.is_non_overlapping_and_dense()) {
+            // (1)
+            if (variable.strides() == new_grad.strides()) {
+              update_grad(new_grad.clone(at::MemoryFormat::Preserve));
+            } else {
+              // Does this dicey-looking sequence attach updated grad to new_grad's
+              // history if GradMode::is_enabled()?  Yes, and @alband says it should.
+              update_grad(std::move(at::empty_strided(variable.sizes(), variable.strides(),
+                                    variable.options().memory_format(c10::nullopt))
```

Comment:
Can you confirm that the effect of the `.memory_format(c10::nullopt)` is to remove the memory format from the variable options?
If so why do you want to remove it? And what will be the layout that the new Tensor gets?

Reply:
> Can you confirm that the effect of the .memory_format(c10::nullopt) is to remove the memory format from the variable options?

I believe so (see [TensorOptions::memory_format](https://github.com/pytorch/pytorch/blob/385165ec674b764eb42ffe396f98fadd08a513eb/c10/core/TensorOptions.h#L223) which calls [set_memory_format](https://github.com/pytorch/pytorch/blob/385165ec674b764eb42ffe396f98fadd08a513eb/c10/core/TensorOptions.h#L505)).

I'm following the pattern [here](https://github.com/pytorch/pytorch/blob/d035d05080729c30636ff30fcc068de3c7e9badd/aten/src/ATen/native/TensorFactories.cpp#L284), removing memory format from the options make sure `empty_strided` actually uses the strides I give it (avoids ambiguity in case `variable.options()` happened to contain a `MemoryFormat` that conflicted with my strides).